src/common/types/core.js

```diff
@@ -26,6 +98,26 @@ type UserMini = {
     type: 'user',
 };
 
+type User = {
```

Comment:
Do we want to dump all of these in `core` or should they each get their own domain-specific file?

Reply:
I think they could probably get split up. I tried to do some splitting, but it's somewhat hard to see the divisions until it's done in some cases.

I was thinking perhaps a file related to the various `BoxItem` related stuff would be good. Any other divisions?

Good with doing this in a followup though?

Reply:
It looks like item, metadata, and user might deserve their own type files, but I think doing it in a follow-up is fine, yeah.